merkle-tree/indexed/src/array.rs

```diff
@@ -0,0 +1,756 @@
+    ///   no node with value greater as 30, so we found it as a one pointing to
+    ///   node 0 (which will always have value 0).
+    /// * The new nullifier is inserted in index 1.
+    /// * `next_*` fields of the low nullifier are updated to point to the new
+    ///   nullifier.
+    ///
+    /// # 2nd insertion
+    ///
+    /// After adding an another value 10, it should look like:
+    ///
+    /// ```
+    /// value      = [ 0]  [30] [10] [0] [0] [0] [0] [0]
+    /// next_index = [ 2]  [ 0] [ 1] [0] [0] [0] [0] [0]
+    /// ```
+    ///
+    /// Because:
+    ///
+    /// * Low nullifier is still the node 0, but this time for differen reason -
+    ///   its `next_index` 2 contains value 30, whish is greater than 10.
+    /// * The new nullifier is inserted as node 2.
+    /// * Low nullifier is pointing to the index 1. We assign the 1st nullifier
+    ///   as the next nullifier of our new nullifier. Therefore, our new nullifier
+    ///   looks like: `[value = 10, next_index = 1]`.
+    /// * Low nullifier is updated to point to the new nullifier. Therefore,
+    ///   after update it looks like: `[value = 0, next_index = 2]`.
+    /// * The previously inserted nullifier, the node 1, remains unchanged.
+    ///
+    /// # 3rd insertion
+    ///
+    /// After adding an another value 20, it should look like:
+    ///
+    /// ```
+    /// value      = [ 0]  [30] [10] [20] [0] [0] [0] [0]
+    /// next_index = [ 2]  [ 0] [ 3] [ 1] [0] [0] [0] [0]
+    /// ```
+    ///
+    /// Because:
+    /// * Low nullifier is the node 2.
+    /// * The new nullifier is inserted as node 3.
+    /// * Low nullifier is pointing to the node 2. We assign the 1st nullifier
+    ///   as the next nullifier of our new nullifier. Therefore, our new
+    ///   nullifier looks like:
+    /// * Low nullifier is updated to point to the new nullifier. Therefore,
+    ///   after update it looks like: `[value = 10, next_index = 3]`.
+    ///
+    /// # 4th insertion
+    ///
+    /// After adding an another value 50, it should look like:
+    ///
+    /// ```
+    /// value      = [ 0]  [30] [10] [20] [50] [0] [0] [0]
+    /// next_index = [ 2]  [ 4] [ 3] [ 1] [0 ] [0] [0] [0]
+    /// ```
+    ///
+    /// Because:
+    ///
+    /// * Low nullifier is the node 1 - there is no node with value greater
+    ///   than 50, so we found it as a one having 0 as the `next_value`.
+    /// * The new nullifier is inserted as node 4.
+    /// * Low nullifier is not pointing to any node. So our new nullifier
+    ///   is not going to point to any other node either. Therefore, the new
+    ///   nullifier looks like: `[value = 50, next_index = 0]`.
+    /// * Low nullifier is updated to point to the new nullifier. Therefore,
+    ///   after update it looks like: `[value = 30, next_index = 4]`.
+    #[test]
+    fn test_append() {
+        let mut nullifier_merkle_tree: IndexingArray<Poseidon, 8> = IndexingArray::default();
```

Comment:
naming is a bit confusing.
```suggestion
        let mut indexing_array: IndexingArray<Poseidon, 8> = IndexingArray::default();
```

Reply:
Fixed